lib/ivis_opengl/pieblitfunc.cpp

```diff
@@ -128,18 +127,24 @@ void GFX::updateTexture(const void *image, int width, int height)
 
 void GFX::buffers(int vertices, const GLvoid *vertBuf, const GLvoid *auxBuf)
 {
-	glBindBuffer(GL_ARRAY_BUFFER, mBuffers[VBO_VERTEX]);
-	glBufferData(GL_ARRAY_BUFFER, vertices * mCoordsPerVertex * sizeof(GLfloat), vertBuf, GL_STATIC_DRAW);
+	if (mBuffers[VBO_VERTEX])
+		delete mBuffers[VBO_VERTEX];
+	mBuffers[VBO_VERTEX] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, vertices * mCoordsPerVertex * sizeof(gfx_api::gfxFloat));
+	mBuffers[VBO_VERTEX]->upload(0, vertices * mCoordsPerVertex * sizeof(GLfloat), vertBuf);
 	if (mType == GFX_TEXTURE)
 	{
-		glBindBuffer(GL_ARRAY_BUFFER, mBuffers[VBO_TEXCOORD]);
-		glBufferData(GL_ARRAY_BUFFER, vertices * 2 * sizeof(GLfloat), auxBuf, GL_STATIC_DRAW);
+		if (mBuffers[VBO_TEXCOORD])
+			delete mBuffers[VBO_TEXCOORD];
+		mBuffers[VBO_TEXCOORD] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, vertices * 2 * sizeof(gfx_api::gfxFloat));
+		mBuffers[VBO_TEXCOORD]->upload(0, vertices * 2 * sizeof(GLfloat), auxBuf);
 	}
 	else if (mType == GFX_COLOUR)
 	{
+		if (mBuffers[VBO_TEXCOORD])
+			delete mBuffers[VBO_TEXCOORD];
+		mBuffers[VBO_TEXCOORD] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, vertices * 4 * sizeof(GLbyte));
```

Comment:
This is not equivalent code to the original code, and will introduce a performance regression. You are creating and deleting the buffer every time through. There is overhead to that.